ui/src/components/Tasks/AddSample.js

```diff
@@ -1,138 +1,138 @@
-                }}
-                col1="4"
-                col2="6"
+          <Form.Group as={Row} className="mb-3" controlId="sampleName">
+            <Col sm={4}>
+              <Form.Label column>Sample name</Form.Label>
+            </Col>
+            <Col sm={8}>
+              <Form.Control
+                type="text"
+                {...register('sampleName', {
+                  required: REQUIRED_MSG,
+                  pattern: { value: PATTERN, message: PATTERN_MSG },
+                })}
+                isValid={isSubmitted && !errors.sampleName}
+                isInvalid={isSubmitted && !!errors.sampleName}
               />
-              <InputField
-                propName="proteinAcronym"
-                label="Protein Acronym"
-                ref={(ref) => {
-                  this.proteinAcronym = ref;
-                }}
-                col1="4"
-                col2="6"
-                onKeyPress={this.handleKeyPress}
+              <Form.Control.Feedback type="invalid">
+                {errors.sampleName?.message}
+              </Form.Control.Feedback>
+            </Col>
+          </Form.Group>
+          <Form.Group as={Row} className="mb-3" controlId="proteinAcronym">
+            <Col sm={4}>
+              <Form.Label column>Protein acronym</Form.Label>
+            </Col>
+            <Col sm={8}>
+              <Form.Control
+                type="text"
+                {...register('proteinAcronym', {
+                  required: REQUIRED_MSG,
+                  pattern: { value: PATTERN, message: PATTERN_MSG },
+                })}
+                isValid={isSubmitted && !errors.proteinAcronym}
+                isInvalid={isSubmitted && !!errors.proteinAcronym}
               />
-            </FieldsRow>
-          </Form>
+              <Form.Control.Feedback type="invalid">
+                {errors.proteinAcronym?.message}
+              </Form.Control.Feedback>
+            </Col>
+          </Form.Group>
         </Modal.Body>
         <Modal.Footer>
           <ButtonToolbar>
-            <Button
-              variant="outline-secondary"
-              disabled={this.props.invalid}
-              onClick={this.addAndEnqueue}
-            >
-              Add Sample
+            <Button className="me-3" type="submit">
+              Mount
             </Button>
             <Button
-              className="ms-3"
               variant="outline-secondary"
-              disabled={this.props.invalid}
-              onClick={this.addAndMount}
+              onClick={handleSubmit(addAndQueue)}
             >
-              Add and mount sample
+              Queue
             </Button>
           </ButtonToolbar>
         </Modal.Footer>
-      </Modal>
-    );
-  }
+      </Form>
+    </Modal>
+  );
 }
 
-AddSample = reduxForm({
-  form: 'addsample',
-  validate,
-})(AddSample);
-
-AddSample = connect((state) => ({
-  initialValues: { ...state.taskForm.taskData.parameters },
```

Comment:
Correct me if I'm wrong, but I don't think the form needs to be pre-filled with any of these parameters.